### ISSUE TITLE
[Support]: 

### ISSUE THREAD

**Issue by Jon811** [blakeblackshear/frigate#7436]
### Describe the problem you are having

Frigate stops record after some time, camera TAPO C100, WiFi, but if i open live in cameras menu, it goes. Looks like ffmpeg stucks after 2-3 hours, sometimes 8.

### Version

0.12.1-367d724

### Frigate config file

```yaml
mqtt:
  enabled: True
  host: core-mosquitto
  port: 1883
  topic_prefix: frigate
  client_id: frigate
  user: user
  password: passwd

record:
  enabled: true
  expire_interval: 10
  retain:
    days: 2
    mode: all
go2rtc:
  streams:
    rtsp_cam: 
      - tapo://Misha2005@192.168.0.49
      - "ffmpeg:rtsp_cam#audio=aac"
cameras:
  rtsp_cam:
    ffmpeg:
      output_args:
        record: preset-record-generic-audio-copy
      inputs:
        - path: rtsp://127.0.0.1:8554/rtsp_cam # <--- the name here must match the name of the camera in restream
          input_args: preset-rtsp-restream
          roles:
            - record
```

### Relevant log output

```shell
Frigate log:
2023-08-10 08:29:59.443687033  [INFO] Starting Frigate...
2023-08-10 08:30:13.440086424  [2023-08-10 11:30:13] frigate.app                    INFO    : Starting Frigate (0.12.1-367d724)
2023-08-10 08:30:13.589490038  [2023-08-10 11:30:13] frigate.app                    INFO    : Creating directory: /tmp/cache
2023-08-10 08:30:13.613809641  [2023-08-10 11:30:13] peewee_migrate                 INFO    : Starting migrations
2023-08-10 08:30:13.744390011  [2023-08-10 11:30:13] peewee_migrate                 INFO    : There is nothing to migrate
2023-08-10 08:30:13.963331170  [2023-08-10 11:30:13] detector.cpu                   INFO    : Starting detection process: 227
2023-08-10 08:30:14.002022559  [2023-08-10 11:30:13] frigate.app                    INFO    : Output process started: 229
2023-08-10 08:30:14.113403622  [2023-08-10 11:30:14] frigate.app                    INFO    : Camera processor started for rtsp_cam: 233
2023-08-10 08:30:14.191064384  [2023-08-10 11:30:14] frigate.app                    INFO    : Capture process started for rtsp_cam: 234
2023-08-10 08:30:14.406957249  [2023-08-10 11:30:13] frigate.detectors              WARNING : CPU detectors are not recommended and should only be used for testing or for trial purposes.
2023-08-10 09:23:04.592416394  [2023-08-10 12:23:04] watchdog.rtsp_cam              INFO    : rtsp_cam exceeded fps limit. Exiting ffmpeg...
2023-08-10 09:23:04.597808291  [2023-08-10 12:23:04] watchdog.rtsp_cam              INFO    : Waiting for ffmpeg to exit gracefully...
2023-08-10 09:23:40.854505404  [2023-08-10 12:23:40] watchdog.rtsp_cam              INFO    : FFmpeg did not exit. Force killing...
2023-08-10 09:23:40.862394413  [2023-08-10 12:23:40] frigate.video                  ERROR   : rtsp_cam: Unable to read frames from ffmpeg process.
2023-08-10 09:23:40.862401985  [2023-08-10 12:23:40] frigate.video                  ERROR   : rtsp_cam: ffmpeg process is not running. exiting capture thread...
2023-08-10 09:23:45.871578578  [2023-08-10 12:23:45] frigate.record                 WARNING : Failed to probe corrupt segment /tmp/cache/rtsp_cam-20230810122301.mp4: 0 - b'[mov,mp4,m4a,3gp,3g2,mj2 @ 0x56269a5ba6c0] moov atom not found\n/tmp/cache/rtsp_cam-20230810122301.mp4: Invalid data found when processing input\n'
2023-08-10 09:23:45.871587124  [2023-08-10 12:23:45] frigate.record                 WARNING : Discarding a corrupt recording segment: /tmp/cache/rtsp_cam-20230810122301.mp4

go2rtc log:
2023-08-10 08:29:59.273720328  [INFO] Preparing go2rtc config...
2023-08-10 08:30:00.085573166  [INFO] Got IP address from supervisor: 192.168.0.130
2023-08-10 08:30:00.522954157  [INFO] Got WebRTC port from supervisor: 8555
2023-08-10 08:30:04.682990652  [INFO] Starting go2rtc...
2023-08-10 08:30:05.495932228  11:30:05.486 INF go2rtc version 1.2.0 linux/amd64
2023-08-10 08:30:05.495939934  11:30:05.495 INF [api] listen addr=:1984
2023-08-10 08:30:05.502122379  11:30:05.501 INF [rtsp] listen addr=:8554
2023-08-10 08:30:05.506973915  11:30:05.503 INF [srtp] listen addr=:8443
2023-08-10 08:30:05.515384687  11:30:05.515 INF [webrtc] listen addr=:8555
2023-08-10 08:30:09.418201794  [INFO] Starting go2rtc healthcheck service...
2023-08-10 08:32:49.453202017  WARNING: mpegts: buffer overflow
2023-08-10 08:33:01.460150413  WARNING: mpegts: buffer overflow
2023-08-10 08:33:07.495300715  WARNING: mpegts: buffer overflow
2023-08-10 08:33:11.534616787  WARNING: mpegts: buffer overflow
2023-08-10 08:33:23.537298383  WARNING: mpegts: buffer overflow
~~around 200 same lines~~
2023-08-10 09:39:03.056791922  WARNING: mpegts: buffer overflow
2023-08-10 09:39:05.052611137  WARNING: mpegts: buffer overflow
2023-08-10 09:39:09.080548607  WARNING: mpegts: buffer overflow
2023-08-10 09:41:15.510336015  WARNING: mpegts: buffer overflow
```

### FFprobe output from your camera

```shell
[{"return_code":0,"stderr":"","stdout":{"programs":[],"streams":[{"avg_frame_rate":"15/1","codec_long_name":"H.264/AVC/MPEG-4AVC/MPEG-4part10","height":1080,"width":1920},{"avg_frame_rate":"0/0","codec_long_name":"AAC(AdvancedAudioCoding)"}]}}]
```

### Frigate stats

```json
Don't know how to get it
```

### Operating system

HassOS

### Install method

HassOS Addon

### Coral version

Other

### Network connection

Wireless

### Camera make and model

TAPO C100

### Any other information that may be helpful

I think it has to be some checking of recording flow and restart, if stuck.

**Comment by NickM-27** [comment]
Frigate already has logic to check if the recording has stopped being created. Once this happens, what do you see in /tmp/cache ?

**Comment by Jon811** [comment]
> 

How could i get in tm/tmp/cache of Frigate container. Now i changed router for new one, moved system from KVM VM to Debian12 HassIO, but it still freeze without any answer.
Logs attached. It was stopped already more than 1,5 hours before.

[go2rtc log.txt](https://github.com/blakeblackshear/frigate/files/12346915/go2rtc.log.txt)
[Frigate log.txt](https://github.com/blakeblackshear/frigate/files/12346916/Frigate.log.txt)

**Comment by NickM-27** [comment]
I would suggest updating go2rtc https://docs.frigate.video/configuration/advanced#custom-go2rtc-version

**Comment by Jon811** [comment]
Wow, That's amazingly easy and works. More then 36 hours without stuck. Only stopped record for 1 minute and continued, some problem with connection. Actually not understand, why Frigate use so old version of rtc2go.

**Comment by klonwar** [comment]
After some time the recording in Frigate stopped working. No clips were created. The page with the list of cameras showed a black screen with an error message instead of a preview, but still I was able to preview camera stream inside. The only thing that helped was to restart Frigate. I tried updating go2rtc to the latest version, but it didn't help.

After adding #backchannel=0 the problem disappeared. Maybe this will help someone with the same issue.
Check more info here: https://github.com/AlexxIT/go2rtc/issues/229
